sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -196,7 +197,12 @@ class OrcFileFormat
         val taskAttemptContext = new TaskAttemptContextImpl(taskConf, attemptId)
 
         if (enableVectorizedReader) {
-          val batchReader = new OrcColumnarBatchReader(capacity)
+          val memoryMode = if (sqlConf.offHeapColumnVectorEnabled) {
```

Comment:
```
[info] - SPARK-28156: self-join should not miss cached view *** FAILED *** (216 milliseconds)
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 1888.0 failed 1 times, most recent failure: Lost task 1.0 in stage 1888.0 (TID 2251) (localhost executor driver): java.lang.NullPointerException: Cannot invoke "org.apache.spark.internal.config.ConfigReader.get(String)" because "reader" is null
[info] 	at org.apache.spark.internal.config.ConfigEntry.readString(ConfigEntry.scala:94)
[info] 	at org.apache.spark.internal.config.FallbackConfigEntry.readFrom(ConfigEntry.scala:270)
[info] 	at org.apache.spark.sql.internal.SQLConf.getConf(SQLConf.scala:5573)
[info] 	at org.apache.spark.sql.internal.SQLConf.offHeapColumnVectorEnabled(SQLConf.scala:5103)
[info] 	at org.apache.spark.sql.execution.datasources.orc.OrcFileFormat.$anonfun$buildReaderWithPartitionValues$2(OrcFileFormat.scala:200)
[info] 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:218)
```
some tests failed

Reply:
Ya, this part fails.